modules/aruco/src/aruco.cpp

```diff
@@ -212,17 +212,20 @@ static void _reorderCandidatesCorners(vector< vector< Point2f > > &candidates) {
 
 
 /**
-  * @brief Check candidates that are too close to each other and remove the smaller one
+  * @brief Check candidates that are too close to each other, save the potential candidates
+  *        (i.e. biggest/smallest contour) and remove the rest
   */
 static void _filterTooCloseCandidates(const vector< vector< Point2f > > &candidatesIn,
-                                      vector< vector< Point2f > > &candidatesOut,
+                                      vector< vector< vector< Point2f > > > &candidatesSetOut,
```

Comment:
> `vector< vector< vector<`

There are really two many nested vectors.
Two values are expected in outer dimension.

Perhaps we should split this parameter into two:
```
vector< vector< Point2f > > &candidatesOut,
vector< vector< Point2f > > &candidatesInvertedOut,
```